src/ccutil/genericvector.h

```diff
@@ -210,7 +210,7 @@ class GenericVector {
   // This function uses memcpy to copy the data, instead of invoking
   // operator=() for each element like double_the_size() does.
   static T* double_the_size_memcpy(int current_size, T* data) {
-    T* data_new = new T[current_size * 2];
+    T* data_new = new T[static_cast<size_t>(current_size) * 2];
```

Comment:
I don't think this is the right longterm solution. Instead of that, size variables like `current_size` should use an unsigned data type, typically `size_t`.

Reply:
I see. Thanks for your advice.

Reply:
@stweil 
I have a question. Changing from one source to the size_t data type and the other to referencing sources encountered the following situation: I have a routine that saves to a file, and I wonder what to do.

1. No modification should be made in this regard.
2. Implement a structure that is compatible with older files.
3. You don't need to make the old file compatible, but you can ignore it after it's upgraded.

I hope to let you know if a solution is in the above category or have a different approach.

ex: (Assume)
If I change type of `used` filed to size_t, I wonder what the effect is.
(https://github.com/tesseract-ocr/tesseract/blob/cede5b34e78ffccc2aa25e19ea80aad31c4354b7/src/ccutil/genericvector.h#L507)

Reply:
For that example changing the type of `used` from `int` to `size_t` would create an incompatible file. Writing to file and reading from file should always use data types with a known size, for example `uint32_t`.

Reply:
@stweil I was impressed with your kind explanation. Thank you always.